createas.c

```diff
@@ -927,12 +1012,33 @@ check_ivm_restriction_walker(Node *node, check_ivm_restriction_context *context)
 				expression_tree_walker(node, check_ivm_restriction_walker, (void *) context);
 				break;
 			}
+		case T_Var:
+			{
+				Var	*variable = (Var *) node;
+				/* If EXISTS subquery refers to vars of the upper query, collect these vars */
+				if (variable->varlevelsup > 0 && context->in_exists_subquery)
+					context->exists_qual_vars = lappend(context->exists_qual_vars, node);
+				break;
+			}
 		case T_SubLink:
 			{
-				ereport(ERROR,
-						(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
-						 errmsg("unsupported subquery on incrementally maintainable materialized view"),
-						 errhint("Only simple subquery in FROM clause is supported.")));
+				/* Now, EXISTS clause is supported only */
```

Comment:
Maybe, "Currently, only EXISTS clause is allowed here" or something like this?

Reply:
Thanks, that's a right.
I will add its comments.

Reply:
I'm sorry I didn't make it clear enough, but I just suggested that "Now, EXISTS clause is ..." at line 1031 should be replaced with "Currently, EXISTS clause is ...", instead of adding the comment here.

Reply:
I understood it. i will fix it